crates/rattler_pypi_interop/src/types/rfc822ish.rs

```diff
@@ -0,0 +1,119 @@
+// Implementation comes from https://github.com/njsmith/posy/blob/main/src/vocab/rfc822ish.rs
+// Licensed under MIT or Apache-2.0
+use std::collections::HashMap;
+use std::str::FromStr;
+
+pub type Fields = HashMap<String, Vec<String>>;
+
+#[cfg_attr(test, derive(Debug, serde::Deserialize, PartialEq, Eq))]
+pub struct RFC822ish {
```

Comment:
I think uv uses [`mailparse`](https://crates.io/crates/mailparse) for this. Id much rather use an existing crate than maintain this ourselves.